subsys/bluetooth/mesh/scheduler_internal.h

```diff
@@ -40,10 +40,16 @@ static inline void scheduler_action_unpack(struct net_buf_simple *buf,
 	entry->scene_number = net_buf_simple_pull_le16(buf);
 }
 
-static inline void scheduler_action_pack(struct net_buf_simple *buf,
+static inline int scheduler_action_pack(struct net_buf_simple *buf,
 				uint8_t idx,
 				const struct bt_mesh_schedule_entry *entry)
 {
+	if (entry->year > 127 || entry->day > 31 || entry->hour > 31 ||
+	    entry->minute > 63 || entry->second > 63 || idx > 15 ||
+		entry == NULL) {
+		return 1;
```

Comment:
Please either return an actual error code, or change to bool (where success == true). In this case for this internal function I think bool (success == true, fail == false) is best.